File: 0392-is-subsequence/0392-is-subsequence.go
func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	curSubCharInd := 0
	for ind := 0; ind < len(t); ind++ {
		if curSubCharInd >= len(s) {
			return true
		}
		if (len(s) - curSubCharInd) > (len(t) - ind) {
			return false
		}
		if t[ind] == s[curSubCharInd] {
			curSubCharInd += 1
		}
	}
	if curSubCharInd == len(s) {
		return true
	}
	return false
}

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0 :
            return True
        if len(t) < len(s):
            return False
        p = 0

        for char in t:
            if char == s[p]:
                p += 1
                
            if p == len(s):
                return True
            
        return False